Index: contrib/yaws/src/json.erl
===================================================================
--- contrib/yaws/src/json.erl	(revision 1738)
+++ contrib/yaws/src/json.erl	(working copy)
@@ -580,8 +580,8 @@
 is_string([]) -> yes;
 is_string(List) -> is_string(List, non_unicode).
 
-is_string([C|Rest], non_unicode) when C >= 0, C =< 255 -> is_string(Rest, non_unicode);
-is_string([C|Rest], _) when C =< 65000 -> is_string(Rest, unicode);
+is_string([C|Rest], non_unicode) when is_integer(C), C >= 0, C =< 255 -> is_string(Rest, non_unicode);
+is_string([C|Rest], _) when is_integer(C), C>= 0, C =< 65000 -> is_string(Rest, unicode);
 is_string([], non_unicode) -> yes;
 is_string([], unicode) -> unicode;
 is_string(_, _) -> no.
